react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+class Stopwatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { on: false, interval: '', counter: 0 };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    if (this.state.on) {
+      clearInterval(this.state.interval);
+      this.setState({ on: false });
+    } else {
+      const stopInterval = setInterval(() => {
+        this.setState({ counter: this.state.counter + 1 });
+      }, 1000);
+      this.setState({ on: true, interval: stopInterval });
+    }
+  }
+
+  render() {
+    if (this.state.on) {
+      return <div className='container'>
+        <h1 className="watch">{this.state.counter}</h1>
+        <i onClick={this.handleClick} className="fa-solid fa-pause fa-4x"></i>
+      </div>;
+    } else {
+      return <div className='container'>
+        <h1 className="watch">{this.state.counter}</h1>
+        <i onClick={this.handleClick} className="fa-solid fa-play fa-4x"></i>
+      </div>;
+    }
```

Comment:
Note the only thing that changes between the two returns is the class on the `i` tag. So a better solution would be:
```JSX
let icon = 'fa-play';

if (this.state.on) {
  icon = 'fa-pause';
}

return (
  <div className='container'>
    <h1 className="watch">{this.state.counter}</h1>
    <i onClick={this.handleClick} className={`fa-solid ${icon} fa-4x`}></i>
  </div>
);
```